python/graphstorm/inference/np_infer.py

```diff
@@ -157,8 +158,10 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
                     if node_id_mapping_file else None
             shuffled_preds = {}
             for ntype, pred in preds.items():
-                assert ntype in infer_ntypes, \
-                    f"{ntype} is not in the set of evaluation ntypes {infer_ntypes}"
+                if ntype not in infer_ntypes:
+                    warnings.warn(f"{ntype} is not in the evaluation ntypes {infer_ntypes}, "
+                                  f"will not do the remapping")
+                    continue
```

Comment:
The DGLBlock problem as mentioned offline.

Reply:
Because here we `continue`, should the behavior is not remapping either saving ntype prediction?
If yes, may change the warning message to `will not do the remapping and saving {ntype} node predictions`.

Reply:
Good callout. Make a change on that.